### ISSUE TITLE
allow shell expansion when creating new file with names like `foo/+{layout,page}.svelte`

### ISSUE THREAD

**Issue by BLucky-gh** [zed-industries/zed#6865]
### Check for existing issues

- [X] Completed

### Describe the feature

currently, zed supports creating parent directories when creating files with `/` in the name, which is convenient, but I think it would also be convenient if the kind of expansion done by shells (as in `{foo,bar}.baz` -> `foo.baz` and `bar.baz`). A good example is making a new route in svelte and making both the page and the layout file at once

**Comment by Moshyfawn** [comment]
Kinda related so I'm going to hijack this feature request by saying I'd like to see the ablity to create nested files by specifying a path as opposed to creating each folder and then the file specifically.

"New File" -> `pages/api/index.ts` would create `index.ts` inside `pages` -> `api` folders, which didn't exist either.

**Comment by BLucky-gh** [comment]
> Kinda related so I'm going to hijack this feature request by saying I'd like to see the ablity to create nested files by specifying a path as opposed to creating each folder and then the file specifically.
> 
> "New File" -> `pages/api/index.ts` would create `index.ts` inside `pages` -> `api` folders, which didn't exist either.

I believe that is already possible. At least I don't remember it not being a thing, but I might have just not tried. If it's not a thing either then def this should be a thing as well

**Comment by Moshyfawn** [comment]
> I believe that is already possible. At least I don't remember it not being a thing, but I might have just not tried. If it's not a thing either then def this should be a thing as well

You are right, my bad 🙈 What I did was click "New Folder" which yielded a folder (as per request lol) but I was expecting it to still create the file instead for some reason. Thanks for the heads-up!
